src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/Datasource.java

```diff
@@ -322,19 +326,31 @@ public boolean isExpired() {
         return Instant.now().isAfter(lastCheckedAt.plus(database.validForInDays, ChronoUnit.DAYS));
     }
 
-    public void setDatabase(final DatasourceManifest datasourceManifest, final String[] fields) {
+    public void setDatabase(final DatasourceManifest datasourceManifest, final List<String> fields) {
         this.database.setProvider(datasourceManifest.getProvider());
         this.database.setMd5Hash(datasourceManifest.getMd5Hash());
         this.database.setUpdatedAt(Instant.ofEpochMilli(datasourceManifest.getUpdatedAt()));
-        this.database.setValidForInDays(database.validForInDays);
-        this.database.setFields(Arrays.asList(fields));
+        this.database.setValidForInDays(datasourceManifest.getValidForInDays());
+        this.database.setFields(fields);
+    }
+
+    public boolean isCompatible(final List<String> fields) {
+        Set<String> fieldsSet = new HashSet<>(fields);
+        for (String field : database.fields) {
+            if (!fieldsSet.contains(field)) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
```suggestion
/**
 * Checks if the database fields are compatible with the given set of input fields.
 *
 * @param inputFields The set of input fields to check for compatibility.
 * @return Returns true if the database fields are compatible with the given input fields, false otherwise.
 */
public boolean isCompatible(final List<String> inputFields) {
    Set<String> fieldsSet = new HashSet<>(inputFields);
    return areFieldsCompatible(fieldsSet, database.fields); 
}

/**
 * Checks if a set of fields is compatible with another set of fields.
 *
 * @param inputFields The set of input fields to check for compatibility.
 * @param databaseFields The set of database fields to check for compatibility.
 * @return Returns true if the database fields are compatible with the given input fields, false otherwise.
 */
private boolean areFieldsCompatible(Set<String> inputFields, Set<String> databaseFields) {
    return inputFields.containsAll(databaseFields);
}
```

Reply:
`database.fields` is a list not set. Let me add a comment though.